intercom-common/src/foreign_ty.rs

```diff
@@ -98,38 +372,37 @@ impl CTypeHandler
                     => panic!( "Fn-types are unsupported." ),
         };
 
-        Some( match ty.as_str() {
-            
-            // Hardcoded handling for parameter types.
-            "ComRc" | "ComItf"
-                => format!( "{}*", self.get_ty(
+        match rust_type.as_str() {
+
+            // Extract a wrapped type.
+            "ComRc" | "ComItf" | "ComResult"
+                => TypeInfoResolver::pass_by_option( TypeInfoResolver::from_type(
                         krate,
                         match **args.unwrap().first().unwrap().value() {
                             syn::GenericArgument::Type( ref t ) => t,
                             _ => return None,
-                        } )? ),
-            "RawComPtr" => "*void".to_owned(),
-            "String" | "BStr" => "BSTR".to_owned(),
-            "usize" => "size_t".to_owned(),
-            "u64" => "uint64".to_owned(),
-            "i64" => "int64".to_owned(),
-            "u32" => "uint32".to_owned(),
-            "i32" => "int32".to_owned(),
-            "u16" => "uint16".to_owned(),
-            "i16" => "int16".to_owned(),
-            "u8" => "uint8".to_owned(),
-            "i8" => "int8".to_owned(),
-            "f64" => "double".to_owned(),
-            "f32" => "float".to_owned(),
-            "c_void" => "void".to_owned(),
-
-            // Default handling checks if we need to rename the type, such as
-            // Foo -> IFoo for implicit interfaces.
-            t => self.get_name_for_ty( krate, t ),
-        } )
-    }
-
-    fn get_name_for_ty(
+                        } ), PassBy::Ptr ),
+
+            // Bare type.
+            _t => Some( TypeInfoResolver::new( krate, RustType::Ident( &segment.ident ) ) ),
+        }
+    }
+
+    /// Determines if the given type is mutable
+    fn is_mutable(
+        mutability: &Option<syn::token::Mut>
+    ) -> bool
+    {
+        match mutability {
+            &Some( _ ) => true,
+            &None => false,
+        }
```

Comment:
```rust
mutability.is_some()
```

Reply:
As a newcomer to rust your suggestion looks more confusing, actually.

Reply:
Is that in response to the right comment?

Mainly wondering because there are a lot of suggestions which I can agree might be confusing here, such as the use of `?`, etc. but I don't think this is one of them. I feel you'd end up with more or less the same code in C#, etc. as well.

```rust
fn is_mutable( mut_token: &Option<..> ) -> bool { mut_token.is_some() }
```

```c#
bool IsVirtual( VirtualToken? virtualToken ) => virtualToken.HasValue
```

I'm not saying you can't be confused by this. I'm more interested in what makes this confusing specifically in Rust. :)